### ISSUE TITLE
Add support for referencing remote WAF policy templates by URL

### ISSUE THREAD

**Issue by olegmukhin** [F5Networks/terraform-provider-bigip#747]
### Is your feature request related to a problem? Please describe.
When creating a WAF policy (bigip_waf_policy resource) a [template_name](https://registry.terraform.io/providers/F5Networks/bigip/latest/docs/resources/bigip_waf_policy#template_name) argument is required to identify the name of a built-in (F5 provided) WAF policy template from a list of allowed values (e.g. [POLICY_TEMPLATE_API_SECURITY](https://clouddocs.f5.com/products/waf-declarative-policy/declarative_policy_v16_1.html#template:~:text=POLICY_TEMPLATE_API_SECURITY)).

Current provider does not allow us to reference a custom WAF policy template created by our organisation stored on a remote system instead of a built-in policy. Functionality to reference external WAF policy templates (by URL or local path) was introduced in BIG-IP v16.1 (see [release notes](https://techdocs.f5.com/kb/en-us/products/big-ip_ltm/releasenotes/product/relnote-bigip-16-1-0.html#rn_asm_1610_new:~:text=whitelist%2Dips.txt%22%20%7D%20%7D%20%7D-,Use%20External%20References%20for%20Base%20Template,-When%20creating%20an)).

This functionality is crucial for adoption of Terraform to manage our F5 estate as we need to maintain our own customised WAF policy templates.

### Describe the solution you'd like
We would like to be able to provide a URL to a remote location of the WAF policy template (e.g. https://example.com/our-waf-template.json) in the template_name argument of the bigip_waf_policy resource to allow the F5 BIG-IP to fetch the template from a remote location during deployment.

One approach could be to detect a URL provided in the template_name argument (i.e. detect a string that starts with `http://` or `https://`) and send it to the BIG-IP as the value for "link" (in the declarative WAF output of the Terraform provider) as per the example provided in the [BIG-IP v16.1 release notes](https://techdocs.f5.com/kb/en-us/products/big-ip_ltm/releasenotes/product/relnote-bigip-16-1-0.html#rn_asm_1610_new:~:text=%7B%20%22link%22%3A%20%22http%3A//172.29.38.211/link/MINIMAL_TEMPLATE.json%22%20%7D) - `"template": { "link": template_name }` instead of as the value for "name" as is currently done for all values of template_name `"template": { "name": template_name }`.

### Describe alternatives you've considered
We do not see any other way to achieve this outcome at this time.

### Additional context
The described functionality of referencing remote WAF policy templates also requires better documentation in the [BIG-IP Declarative WAF schema](https://clouddocs.f5.com/products/waf-declarative-policy/declarative_policy_v16_1.html#template) as it only appears to be documented in the [BIG-IP v16.1 release notes](https://techdocs.f5.com/kb/en-us/products/big-ip_ltm/releasenotes/product/relnote-bigip-16-1-0.html#rn_asm_1610_new:~:text=%7B%20%22link%22%3A%20%22http%3A//172.29.38.211/link/MINIMAL_TEMPLATE.json%22%20%7D).

**Comment by olegmukhin** [comment]
To further clarify, our current WAF template contains a number of security features that are not yet implemented in the Terraform provider (e.g. custom signature sets created using filters, HTTP compliance settings, etc.) so we were hoping to emulate the behaviour of the below declarative WAF policy example where we could have the provider import an external template which contains these additional security settings.

```json
{
    "policy": {
        "name": "Example Policy Name",
        "description": "This is an example policy that fetched remote template.",
        "template": {
            "link": "https://example.com/our-waf-template.json"
        }
```

**Comment by KrithikaChidambaram** [comment]
Hi, this is being tracked internally: INFRAANO-1029.
Thanks!

**Comment by olegmukhin** [comment]
To clarify, after further discussions it seems that a better approach than passing a template URL to the F5 is to change the logic of how the policy_import_json file is handled.

If we explicitly set arguments/inputs of the bigip_waf_policy then these should overwrite the values in the JSON file provided through policy_import_json. This will allow a template (starting point) to be provided in the policy_import_json.

**Comment by olegmukhin** [comment]
Closed by https://github.com/F5Networks/terraform-provider-bigip/pull/759

**Comment by olegmukhin** [comment]
@RavinderReddyF5 @KrithikaChidambaram After testing the new template_link argument I can see that the template_name is still mandatory. In our testing when both the template_name and template_link are set the F5 uses the template_name and ignores the template_link. Therefore, it is not possible to provide the template URL to the F5 in way where it will actually use it. Therefore this issue should still be open.

**Comment by RavinderReddyF5** [comment]
Hi @olegmukhin i believe adding support with `policy_import_json` with dumping full json onto BIG-IP will resolves alternatively for requirement of template_link URL. please see if that helps for your requirement? do we have any use case where it does not help with `policy_import_json`.

**Comment by olegmukhin** [comment]
@RavinderReddyF5 sorry I thought template_link was now also supported as I saw it in latest release. We can test using policy_import_json further but are unable to do this due to a bug we've encountered (see #766).

Only use case that we need to understand better where policy_import_json might not be sufficient is when a Terraform module is created to define a standard way of deploying a WAF policy. This is something we think is important.

In such a scenario you would have the module pass the template JSON to policy_import_json, which would work, but you then would not have a way of actually importing a WAF policy due to configuration drift.
